HousingFinanceInterimApi/V1/UseCase/GenerateRentPositionUseCase.cs

```diff
@@ -71,28 +70,57 @@ public async Task<StepResponse> ExecuteAsync()
                     LoggingHandler.LogInfo($"Folder ID: {googleFileSetting.GoogleIdentifier}");
                     LoggingHandler.LogInfo($"File count: {folderFiles.Count}");
 
-                    LoggingHandler.LogInfo($"Deleting old files");
-                    foreach (var file in folderFiles.Where(f => f.Name.Equals(fileName)).ToList())
-                    {
-                        await _googleClientService.DeleteFileInDrive(file.Id).ConfigureAwait(false);
-                    }
-
                     var isSuccess = await _googleClientService.UploadCsvFile(rentPosition, fileName, googleFileSetting.GoogleIdentifier)
                         .ConfigureAwait(false);
 
                     if (!isSuccess)
                         throw new Exception("Failed to upload to Rent Position folder (Qlik)");
 
+                    LoggingHandler.LogInfo($"Deleting old files");
+                    foreach (var file in folderFiles.Where(f => f.Name.Equals(fileName)).ToList())
+                    {
+                        await _googleClientService.DeleteFileInDrive(file.Id).ConfigureAwait(false);
+                    }
                 }
 
                 googleFileSettings = await GetGoogleFileSetting(_rentPositionBkpLabel).ConfigureAwait(false);
                 foreach (var googleFileSetting in googleFileSettings)
                 {
                     var isSuccess = await _googleClientService.UploadCsvFile(rentPosition, $"{DateTime.Now:yyyyMMdd_HHmmss}.csv",
                         googleFileSetting.GoogleIdentifier).ConfigureAwait(false);
+                    var folderFiles = await _googleClientService.GetFilesInDriveAsync(googleFileSetting.GoogleIdentifier)
+                        .ConfigureAwait(false);
```

Comment:
Weren't the `folderFiles` already set on line **67**?
![image](https://github.com/LBHackney-IT/housing-finance-interim-api/assets/43747286/3b3c9b73-eb31-4b78-b4e2-4e7066ebc3ff)